packages/ai/src/ai/web/metrics/tracing.py

```
@@ -0,0 +1,131 @@
+    Returns:
+        An OpenTelemetry ``Tracer`` instance.
+    """
+    return trace.get_tracer(name)
+
+
+def shutdown_tracing() -> None:
+    """Flush pending spans and shut down the tracer provider.
+
+    Safe to call even if ``setup_tracing`` was never invoked.
+    """
+    global _tracer_provider
```

Ruff (0.15.7)

[warning] Using the global statement to update `_tracer_provider` is discouraged

(PLW0603)